File: Samples/AllJoyn/ProducerExperiences/cs/AppData.cs
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

using org.alljoyn.Onboarding;
using System;
using System.Collections.Generic;
using Windows.Storage;

namespace AllJoynProducerExperiences
{
    class AppData
    {
        private const string isUpperCaseEnabledPropertyKey = "IsUpperCaseEnabled";
        private const string onboardingConfigureSsidKey = "OnboardingConfigureSsid";
        private const string onboardingConfigurePassphraseKey = "OnboardingConfigurePassphrase";
        private const string onboardingConfigureAuthenticationTypeKey = "OnboardingConfigureAuthenticationType";
        private const string onboardingConfigurationStateKey = "OnboardingConfigurationState";
        private const string onboardingLastErrorCodeKey = "OnboardingLastErrorCode";
        private const string onboardingLastErrorMessageKey = "OnboardingLastErrorMessage";
        private static ApplicationDataContainer localSettings = ApplicationData.Current.LocalSettings;

        public static event EventHandler IsUpperCaseEnabledPropertyDataChanged;

        public static bool IsUpperCaseEnabled
        {
            get
            {
                if (!localSettings.Values.ContainsKey(isUpperCaseEnabledPropertyKey))
                {
                    localSettings.Values[isUpperCaseEnabledPropertyKey] = false;
                }
                return (bool)localSettings.Values[isUpperCaseEnabledPropertyKey];
            }
            set
            {
                if (IsUpperCaseEnabled != value)
                {
                    localSettings.Values[isUpperCaseEnabledPropertyKey] = value;
                    EventHandler handler = IsUpperCaseEnabledPropertyDataChanged;
                    if (handler != null)
                    {
                        handler(new object(), new EventArgs());
                    }
                }
            }
        }

        public static string OnboardingConfigureSsid
        {
            get
            {
                if (!localSettings.Values.ContainsKey(onboardingConfigureSsidKey))
                {
                    return null;
                }
                else
                {
                    return (string)localSettings.Values[onboardingConfigureSsidKey];
                }
            }
            set
            {
                localSettings.Values[onboardingConfigureSsidKey] = value;
            }
        }

        public static string OnboardingConfigurePassphrase
        {
            get
            {
                if (!localSettings.Values.ContainsKey(onboardingConfigurePassphraseKey))
                {
                    return null;
                }
                else
                {
                    return (string)localSettings.Values[onboardingConfigurePassphraseKey];
                }
            }
            set
            {
                localSettings.Values[onboardingConfigurePassphraseKey] = value;
            }
        }

        public static short OnboardingConfigureAuthType
        {
            get
            {
                if (!localSettings.Values.ContainsKey(onboardingConfigureAuthenticationTypeKey))
                {
                    return (short)OnboardingAuthenticationType.Any;
                }
                else
                {
                    return (short)localSettings.Values[onboardingConfigureAuthenticationTypeKey];
                }
            }
            set
            {
                localSettings.Values[onboardingConfigureAuthenticationTypeKey] = value;
            }
        }

        public static short OnboardingConfigurationState
        {
            get
            {
                if (!localSettings.Values.ContainsKey(onboardingConfigurationStateKey))
                {
                    localSettings.Values[onboardingConfigurationStateKey] = (short)ConfigurationState.NotConfigured;
                }
                return (short)localSettings.Values[onboardingConfigurationStateKey];
            }
            set
            {
                localSettings.Values[onboardingConfigurationStateKey] = value;
            }
        }

        public static short LastErrorCode
        {
            get
            {
                if (localSettings.Values.ContainsKey(onboardingLastErrorCodeKey))
                {
                    return (short)localSettings.Values[onboardingLastErrorCodeKey];
                }
                else
                {
                    return (short)ConnectionResultCode.ErrorMessage;
                }
            }
            set
            {
                localSettings.Values[onboardingLastErrorCodeKey] = value;
            }
        }

        public static string LastErrorMessage
        {
            get
            {
                if (localSettings.Values.ContainsKey(onboardingLastErrorMessageKey))
                {
                    return (string)localSettings.Values[onboardingLastErrorMessageKey];
                }
                else
                {
                    return "No Last Error available";
                }
            }
            set
            {
                localSettings.Values[onboardingLastErrorMessageKey] = value;
            }
        }

        public static List<OnboardingScanListItem> SampleNetworkScanList
        {
            get
            {
                // Simulated network scan data.

                OnboardingScanListItem network1 = new OnboardingScanListItem();
                network1.Value1 = "SampleNetwork1";
                network1.Value2 = (short)OnboardingAuthenticationType.Open;

                OnboardingScanListItem network2 = new OnboardingScanListItem();
                network2.Value1 = "SampleNetwork2";
                network2.Value2 = (short)OnboardingAuthenticationType.Wep;

                OnboardingScanListItem network3 = new OnboardingScanListItem();
                network3.Value1 = "SampleNetwork3";
                network3.Value2 = (short)OnboardingAuthenticationType.WpaTkip;

                OnboardingScanListItem network4 = new OnboardingScanListItem();
                network4.Value1 = "SampleNetwork4";
                network4.Value2 = (short)OnboardingAuthenticationType.WpaCcmp;

                OnboardingScanListItem network5 = new OnboardingScanListItem();
                network5.Value1 = "SampleNetwork5";
                network5.Value2 = (short)OnboardingAuthenticationType.Wpa2Tkip;

                OnboardingScanListItem network6 = new OnboardingScanListItem();
                network6.Value1 = "SampleNetwork6";
                network6.Value2 = (short)OnboardingAuthenticationType.Wpa2Ccmp;

                OnboardingScanListItem network7 = new OnboardingScanListItem();
                network7.Value1 = "SampleNetwork7";
                network7.Value2 = (short)OnboardingAuthenticationType.Wps;

                List<OnboardingScanListItem> networkScanList = new List<OnboardingScanListItem>();
                networkScanList.Add(network1);
                networkScanList.Add(network2);
                networkScanList.Add(network3);
                networkScanList.Add(network4);
                networkScanList.Add(network5);
                networkScanList.Add(network6);
                networkScanList.Add(network7);

                return networkScanList;
            }
        }

        public const string SampleNetworkPassword = "SamplePassword";
    }
}

File: Samples/AllJoyn/ProducerExperiences/js/js/appdata.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {

    "use strict";

    var localSettings = Windows.Storage.ApplicationData.current.localSettings;
    var onboarding = org.alljoyn.Onboarding;

    var isUpperCaseEnabledPropertyKey = "IsUpperCaseEnabled";
    var onboardingConfigureSsidKey = "OnboardingConfigureSsid";
    var onboardingConfigurePassphraseKey = "OnboardingConfigurePassphrase";
    var onboardingConfigureAuthenticationTypeKey = "OnboardingConfigureAuthenticationType";
    var onboardingConfigurationStateKey = "OnboardingConfigurationState";
    var onboardingLastErrorCodeKey = "OnboardingLastErrorCode";
    var onboardingLastErrorMessageKey = "OnboardingLastErrorMessage";

    var secureInterfaceProperties = {};

    Object.defineProperty(secureInterfaceProperties, 'isUpperCaseEnabled', {
        get: function () {
            if (!localSettings.values.hasKey(isUpperCaseEnabledPropertyKey)) {
                localSettings.values[isUpperCaseEnabledPropertyKey] = false;
            }
            return localSettings.values[isUpperCaseEnabledPropertyKey];
        },
        set: function (value) {
            localSettings.values[isUpperCaseEnabledPropertyKey] = value;
        }
    });

    var onboardingConfiguration = {};

    Object.defineProperty(onboardingConfiguration, 'ssid', {
        get: function () {
            if (!localSettings.values.hasKey(onboardingConfigureSsidKey)) {
                return null;
            } else {
                return localSettings.values[onboardingConfigureSsidKey];
            }
        },
        set: function (value) {
            localSettings.values[onboardingConfigureSsidKey] = value;
        }
    });

    Object.defineProperty(onboardingConfiguration, 'passphrase', {
        get: function () {
            if (!localSettings.values.hasKey(onboardingConfigurePassphraseKey)) {
                return null;
            } else {
                return localSettings.values[onboardingConfigurePassphraseKey];
            }
        },
        set: function (value) {
            localSettings.values[onboardingConfigurePassphraseKey] = value;
        }
    });

    Object.defineProperty(onboardingConfiguration, 'authenticationType', {
        get: function () {
            if (!localSettings.values.hasKey(onboardingConfigureAuthenticationTypeKey)) {
                return OnboardingEnumTypes.AuthenticationType.Any;
            } else {
                return localSettings.values[onboardingConfigureAuthenticationTypeKey];
            }
        },
        set: function (value) {
            localSettings.values[onboardingConfigureAuthenticationTypeKey] = value;
        }
    });

    var onboardingProperties = {};

    Object.defineProperty(onboardingProperties, 'configurationState', {
        get: function () {
            if (!localSettings.values.hasKey(onboardingConfigurationStateKey)) {
                localSettings.values[onboardingConfigurationStateKey] = OnboardingEnumTypes.ConfigurationState.NotConfigured;
            }
            return localSettings.values[onboardingConfigurationStateKey];
        },
        set: function (value) {
            localSettings.values[onboardingConfigurationStateKey] = value;
        }
    });

    Object.defineProperty(onboardingProperties, 'lastErrorCode', {
        get: function () {
            if (!localSettings.values.hasKey(onboardingLastErrorCodeKey)) {
                return OnboardingEnumTypes.ConnectionResultCode.ErrorMessage;
            } else {
                return localSettings.values[onboardingLastErrorCodeKey];
            }
        },
        set: function (value) {
            localSettings.values[onboardingLastErrorCodeKey] = value;
        }
    });

    Object.defineProperty(onboardingProperties, 'lastErrorMessage', {
        get: function () {
            if (!localSettings.values.hasKey(onboardingLastErrorMessageKey)) {
                return "No Last Error available";
            } else {
                return localSettings.values[onboardingLastErrorMessageKey];
            }
        },
        set: function (value) {
            localSettings.values[onboardingLastErrorMessageKey] = value;
        }
    });

    var getSampleNetworkScanList = function () {
        var network1 = new onboarding.OnboardingScanListItem();
        network1.value1 = "SampleNetwork1";
        network1.value2 = OnboardingEnumTypes.AuthenticationType.Open;

        var network2 = new onboarding.OnboardingScanListItem();
        network2.value1 = "SampleNetwork2";
        network2.value2 = OnboardingEnumTypes.AuthenticationType.Wep;

        var network3 = new onboarding.OnboardingScanListItem();
        network3.value1 = "SampleNetwork3";
        network3.value2 = OnboardingEnumTypes.AuthenticationType.WpaTkip;

        var network4 = new onboarding.OnboardingScanListItem();
        network4.value1 = "SampleNetwork4";
        network4.value2 = OnboardingEnumTypes.AuthenticationType.WpaCcmp;

        var network5 = new onboarding.OnboardingScanListItem();
        network5.value1 = "SampleNetwork5";
        network5.value2 = OnboardingEnumTypes.AuthenticationType.Wpa2Tkip;

        var network6 = new onboarding.OnboardingScanListItem();
        network6.value1 = "SampleNetwork6";
        network6.value2 = OnboardingEnumTypes.AuthenticationType.Wpa2Ccmp;

        var network7 = new onboarding.OnboardingScanListItem();
        network7.value1 = "SampleNetwork7";
        network7.value2 = OnboardingEnumTypes.AuthenticationType.Wps;

        var networkScanList = [];
        networkScanList.push(network1);
        networkScanList.push(network2);
        networkScanList.push(network3);
        networkScanList.push(network4);
        networkScanList.push(network5);
        networkScanList.push(network6);
        networkScanList.push(network7);

        return networkScanList;
    }

    var sampleNetworkPassword = "SamplePassword";

    WinJS.Namespace.define("AppData", {
        secureInterfaceProperties: secureInterfaceProperties,
        onboardingConfiguration: onboardingConfiguration,
        onboardingProperties: onboardingProperties,
        getSampleNetworkScanList: getSampleNetworkScanList,
        sampleNetworkPassword: sampleNetworkPassword
    });
})();